jni/tool/generate_ide_files.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:args/args.dart';
+
+void runCommand(String exec, List<String> args, String workingDirectory) {
+  stderr.writeln('+ $exec ${args.join(" ")}');
+  final process =
+      Process.runSync(exec, args, workingDirectory: workingDirectory);
+  if (process.exitCode != 0) {
+    stdout.writeln(process.stdout);
+    stderr.writeln(process.stderr);
+    throw "command failed with exit code ${process.exitCode}";
+  }
+}
+
+void main(List<String> arguments) {
```

Comment:
Not sure. It's intended to be run on dev setup, instead of a non-portable bash precommit hook.

It assumes lot of one-time stuff is already done. Eg: `flutter build apk` in Android examples. This way it can complete fairly faster on local dev setup.